server/routes/serviceProviderReferrals/reportingForm.test.ts

```diff
@@ -0,0 +1,154 @@
+import TestUtils from '../../../testutils/testUtils'
+import DateUtils from '../../utils/dateUtils'
+import ReportingForm from './reportingForm'
+
+describe(ReportingForm, () => {
+  describe('data', () => {
+    describe('with valid data', () => {
+      it('returns an object with two Calendar Day values', async () => {
+        const request = TestUtils.createRequest({
+          'from-date-year': '2021',
+          'from-date-month': '06',
+          'from-date-day': '12',
+          'to-date-year': '2021',
+          'to-date-month': '06',
+          'to-date-day': '13',
+        })
+
+        const result = await new ReportingForm(request).data()
+
+        expect(result.reportingDateRange!.fromIncludingDate).toEqual('2021-06-12')
```

Comment:
I reckon (see [my comment on #516](https://github.com/ministryofjustice/hmpps-interventions-ui/pull/516#discussion_r662350752)) that these should be `CalendarDay` instances.